docs/clustering/multi-node-setup.rst

```diff
@@ -309,34 +309,16 @@ For example:
     whenever you do configure it, otherwise CrateDB may form multiple
     independent clusters (which may result in data loss).
 
-CrateDB requires a `quorum`_ of nodes before a master can be elected. A quorum
-ensures that the cluster does not elect multiple masters in the event of a
-network partition (also known as a `split-brain`_ scenario).
-
-CrateDB (versions 4.x and above) will automatically determine the ideal `quorum
-size`_. If you are using CrateDB versions 3.x and below, you must manually set
-the quorum size using the `discovery.zen.minimum_master_nodes`_ setting.
-
-.. NOTE::
-
-    For a three-node cluster, CrateDB will set the quorum size to three.
-    Consequentially, you must declare all nodes to be master-eligible. Consult
-    the `quorum guide`_ for detailed information about quorum size
-    considerations.
-
-If you configure fewer master-eligible nodes than the ideal quorum size,
-CrateDB will issue a warning (visible in the logs and the `admin UI`_).
-
-
 .. _metadata-gateway:
 
 Metadata gateway
 ^^^^^^^^^^^^^^^^
 
 When running a multi-node cluster, you must configure the `metadata gateway`_
-so that CrateDB knows how to recover its state. For a three-node cluster, set
-`gateway.expected_data_nodes`_ to **3** and `gateway.recover_after_data_nodes`_
-to **3**.
+so that CrateDB knows how to recover its state. For a three-nodes cluster, set
+`gateway.expected_data_nodes`_ to **3**, `gateway.recover_after_data_nodes`_
+could be either **2** or **3**, only set it to **2** if you plan to have at
+least 1 replica for all your tables.
```

Comment:
I'm not sure if we should document the `2` variant as it may only increase complexity and confusion like: why only 2 if I have replicas configured? It's valid but not sure how much it helps specially for non super experts.

Reply:
I just tested with a table with `number_of_replicas`=0 , in an environment with `expected_data_nodes`=3 and `recover_after_data_nodes`=2. Starting then with 2 nodes, tables for which shards are available work normally, and the table missing shards does not work, but it resumes working as normal once the 3rd node joins back.
Do you know of some circumstances where this would not be safe? if it is safe maybe  the how to should point to the 2 variant only? what I want to pre-empt here is the somehow counterintuitive situation where a 3-nodes cluster, that can withstand without issues the shutdown of one node, fails to start from cold with the same 2 nodes because the current version of the docs recommends setting `recover_after_data_nodes` to match the expected number of nodes.

Reply:
And also when replica's are configured. Because of the replica's, all shards for a working table are available, but the replica shards will turn then into primary shards, the missing original primary and replica shards are treat as gone and maybe new replica shards will be created on different nodes. This will result in network and disk load which one want to avoid if the other nodes are just being restart and are coming back soon.

Reply:
>  it will result in a broken table (RED health) as long as the missing shards aren't available

Yes, but this could be acceptable because we get an error message when trying to query the data that is not available, what I worry about is if there are situations where data may get corrupted or changes lost. Could we for instance risk losing changes made since the last refresh interval plus flush interval?

> in rolling-upgrade or any other node restart scenario, you want the cluster to start the recovery of all tables only if all shards are available

I was focusing on a start from cold scenario (imagine power cut - storage controller fried on one node - I still have all nodes except 1 and my tables have replicas - I need to restore service)
For rolling restarts, I think high availability would still be very desiderable, I think it is one of the main reasons why people go for cluster deployments instead of standalone instances.

> the missing original primary and replica shards are treat as gone and maybe new replica shards will be created on different nodes. This will result in network and disk load which one want to avoid if the other nodes are just being restarted

This is a very good point, I need to add a note about this, it came out in a conversation we had with @proddata  2 weeks ago.
For a start from cold, where some nodes come back online faster than other nodes we have `recover_after_time`, but from my tests I am under the impression this does not help in the rolling upgrade scenario, maybe I am missing something or there is some other setting?

Reply:
> but from my tests I am under the impression this does not help in the rolling upgrade scenario, maybe I am missing something or there is some other setting?

Still not quite understand what the "issue" with rolling upgrades should be? We even refer in our docs to prevent allocations, by only allowing `new_primaries` https://crate.io/docs/crate/howtos/en/latest/admin/rolling-upgrade.html#step-1-disable-allocations which is btw done automatically for cloud / operator managed clusters.

i.e. during a rolling update, nothing unintended would happen.

Reply:
Just think of simple rolling restarts instead of rolling upgrades without using graceful stops or disabled allocations.

Reply:
> Could we for instance risk losing changes made since the last refresh interval plus flush interval?

I tested with `"translog.flush_threshold_size"= 100000000,"refresh_interval" = 10000` (durability set to the default which is `REQUEST`) and no data seems to go lost, when there are replicas records inserted just before the abrupt shutdown are available when nodes with replicas come back online and in the case of no replicas the record are available as soon as the node with the replica joins.

Reply:
I think we are a mixing up many things in this conversation. Recovery and the related settings are **somewhat** independent from replica settings and therefore I would not at such a note to the documentation. 

In fact 2 data nodes could be perfectly fine in a setup of 2 data nodes and 1 master node, with no replicas. What is "right" largely depends on the setup and expectations. The recovery process also is not "free" and might take some time, why one would want to start recovery, even thought not all nodes are available yet.

Reply:
> durability set to the default which is REQUEST

Which ensures that everything gets written to the wal/translog of all shards.

> Just think of simple rolling restarts instead of rolling upgrades without using graceful stops or disabled allocations.

Which is problematic in that sense of that one doesn't know if a write was successful and CrateDB basically auto-commits transactions.

In terms of recovery starting after one node is gone all of a sudden: I don't think it is a big issue. If a node leaves unexpectedly the time how long it takes to come back is most likely also unknown most. Recovery is also delayed by `unassigned.node_left.delayed_timeout` on a table level which defaults to `1m`. So if a node comes back after 2 minutes, recover might have been started, but gets interrupted and the shards on the node get started again.
One could argue for a cluster wide setting instead for comfort reasons.